Sources/Interaction/Manipulators/3DControllerModelSelectorManipulator/index.js

```diff
@@ -0,0 +1,204 @@
+
+    // compute our transform
+    const transform = mat4.create();
+    mat4.translate(transform, transform, worldPosition);
+    mat4.rotate(transform, transform, newOrientationAngle, newOrientationAxis);
+    mat4.translate(transform, transform, vec3.negate([], worldPosition));
+    mat4.translate(transform, transform, translation);
+
+    // multiply our transform by the prop internal matrix
+    mat4.multiply(transform, transform, matrix);
+
+    // Multiply the transform with the current prop orientation to get the delta
+    // that we have to apply to the prop
+    const matRotation = mat4.getRotation([], transform);
+    const propCurrentOrientation = prop.getOrientationQuaternion();
+    const propCurrentOrientationConjugated = quat.conjugate(
+      [],
+      propCurrentOrientation
+    );
+    const propNewOrientation = quat.multiply(
+      [],
+      propCurrentOrientationConjugated,
+      matRotation
+    );
+
+    quat.normalize(propNewOrientation, propNewOrientation);
+
+    const matTranslation = mat4.getTranslation([], transform);
+    const matScaling = mat4.getScaling([], transform);
+
+    // Update the prop internal matrix
+    prop.setPosition(...matTranslation);
+    prop.setScale(...matScaling);
+    prop.rotateQuaternion(propNewOrientation);
+  }
+
+  publicAPI.onButton3D = (
+    _interactorStyle,
+    renderer,
+    _state,
+    _device,
+    _input,
+    pressed,
+    targetPosition,
+    targetOrientation
+  ) => {
+    // If the button is not pressed, clear the state
+    if (!pressed) {
+      model.lastOrientation = null;
+      model.lastWorldPosition = null;
+      pickedProp = null;
+      return;
+    }
+
+    const camera = renderer.getActiveCamera();
+    const physicalToWorldMatrix = [];
+    camera.getPhysicalToWorldMatrix(physicalToWorldMatrix);
+
+    // Since the targetPosition we get is in physical coordinates,
+    // transform it using the physicalToWorldMatrix to get it in woorld coordinates
+    const targetRayWorldPosition = vec3.transformMat4(
+      [],
+      [targetPosition.x, targetPosition.y, targetPosition.z],
+      physicalToWorldMatrix
+    );
+
+    const targetRayWorldDirection = camera.physicalOrientationToWorldDirection([
+      targetOrientation.x,
+      targetOrientation.y,
+      targetOrientation.z,
+      targetOrientation.w,
+    ]);
+
+    const dist = renderer.getActiveCamera().getClippingRange()[1];
+    const rayPoint1 = [...targetRayWorldPosition, 1.0];
+    const rayPoint2 = [
+      rayPoint1[0] - targetRayWorldDirection[0] * dist,
+      rayPoint1[1] - targetRayWorldDirection[1] * dist,
+      rayPoint1[2] - targetRayWorldDirection[2] * dist,
+      1.0,
+    ];
+
+    // Perform picking on the given renderer
+    picker.pick3DPoint(rayPoint1, rayPoint2, renderer);
+    const props = picker.getActors();
+
+    // If we have picked props, store the first one.
+    if (props.length > 0) {
+      pickedProp = props[0];
```

Comment:
should probably call `pick()` on `pickedProp`

Reply:
looks like this function is not implemented. it's going to log an error, does not look ideal.

Reply:
fair enough